Rules/ReviewUnusedParameter.cs

```diff
@@ -97,11 +98,39 @@ public IEnumerable<DiagnosticRecord> AnalyzeScript(Ast ast, string fileName)
                 // find all declared parameters
                 IEnumerable<Ast> parameterAsts = scriptBlockAst.FindAll(oneAst => oneAst is ParameterAst, false);
 
+                // does the scriptblock have a process block where either $PSItem or $_ is referenced
+                bool hasProcessBlockWithPSItemOrUnderscore = false;
+                if (scriptBlockAst.ProcessBlock != null)
+                {
+                    IDictionary<string, int> processBlockVariableCount = GetVariableCount(scriptBlockAst.ProcessBlock);
+                    processBlockVariableCount.TryGetValue("_", out int underscoreVariableCount);
+                    processBlockVariableCount.TryGetValue("psitem", out int psitemVariableCount);
+                    if (underscoreVariableCount > 0 || psitemVariableCount > 0)
+                    {
+                        hasProcessBlockWithPSItemOrUnderscore = true;
+                    }
+                }
+
                 // list all variables
                 IDictionary<string, int> variableCount = GetVariableCount(scriptBlockAst);
 
                 foreach (ParameterAst parameterAst in parameterAsts)
                 {
+                    // Check if the parameter has the ValueFromPipeline attribute
+                    NamedAttributeArgumentAst valueFromPipeline = (NamedAttributeArgumentAst)parameterAst.Find(
+                        valFromPipelineAst => valFromPipelineAst is NamedAttributeArgumentAst namedAttrib && string.Equals(
+                            namedAttrib.ArgumentName, "ValueFromPipeline",
+                            StringComparison.OrdinalIgnoreCase
+                        ),
+                        false
+                    );
+                    // If the parameter has the ValueFromPipeline attribute and the scriptblock has a process block with
+                    // $_ or $PSItem usage, then the parameter is considered used
+                    if (valueFromPipeline?.GetValue() == true && hasProcessBlockWithPSItemOrUnderscore)

```

Comment:
```suggestion
                    if (valueFromPipeline?.GetValue() && hasProcessBlockWithPSItemOrUnderscore)

```

You shouldn't need to explicitly compare this to true in the LHS of compound boolean inside an if conditional (and it confused me). I could be wrong, maybe there's some weirdness with the type returned by `GetValue()` but I'm assuming, given that it's being directly compared to a bool, that it is a bool.

Reply:
Ah yes - this is a strongly typed language with concrete return types, must remember that! 😅

Appreciate your time looking over this, thanks

Reply:
I come from C++ so C# still confuses me 😅 (and don't get me started on TypeScript...)

Reply:
It looks like the latter is more consistent with the codebase so that's what I've gone with. 🤞